src/phpbrowscap/Browscap.php

```diff
@@ -526,8 +526,12 @@ public function updateCache()
     {
         $lockfile = $this->cacheDir . 'cache.lock';
 
-        if (file_exists($lockfile) || !touch($lockfile)) {
-            throw new Exception(sprintf('temporary file %s already exists', $lockfile));
+        $lockRes = fopen($lockfile, 'w+');
+        if (false === $lockRes) {
+            throw new Exception(sprintf('error opening lockfile %s', $lockfile));
+        }
+        if(false === flock($lockRes, LOCK_EX | LOCK_NB)) {
```

Comment:
please write `if (` (with space), please change the other `if`s which you added too